src/sql/workbench/parts/notebook/cellToggleMoreActions.ts

```diff
@@ -141,3 +144,39 @@ export class ClearCellOutputAction extends CellActionBase {
 	}
 
 }
+
+export class RunCellsAction extends CellActionBase {
+	constructor(id: string, label: string, private isAfter: boolean,
+		@INotificationService notificationService: INotificationService,
+		@INotebookService private notebookService: INotebookService,
+	) {
+		super(id, label, undefined, notificationService);
+	}
+
+	public canRun(context: CellContext): boolean {
+		return context.cell && context.cell.cellType === CellTypes.Code;
+	}
+
+	doRun(context: CellContext): Promise<void> {
+		try {
+			let cell = context.cell || context.model.activeCell;
+			if (cell) {
+				let editor = this.notebookService.findNotebookEditor(cell.notebookModel.notebookUri);
+				if (editor) {
+					if (this.isAfter) {
+						editor.runAllCells(cell.id, undefined);
```

Comment:
This is a promise, therefore if we don't await, we will let unhandled promise rejections be thrown which isn't great (and misses the point of your try/catch). Can we make method async & await these so try/catch gets invoked? Thanks!